erpnext/accounts/doctype/invoice_discounting/invoice_discounting.js

```diff
@@ -205,9 +204,15 @@ frappe.ui.form.on('Invoice Discounting', {
 });
 
 frappe.ui.form.on('Discounted Invoice', {
-	sales_invoice: (frm) => {
+	sales_invoice: (frm, cdt, cdn) => {
 		frm.events.calculate_total_amount(frm);
-		frm.events.refresh_filters(frm);
+		frm.events.refresh_filters(frm);	
+		
+		let row = locals[cdt][cdn];
```

Comment:
Remove this. This only works if Invoice Discounting Is done through user interface. Is doesn't work if created from server side or through API. Test Cases failing because of it